r/configure

```diff
@@ -86,7 +86,11 @@ CPPFLAGS=$("${R_HOME}"/bin/R CMD config CPPFLAGS)
 
 # If libarrow uses the old GLIBCXX ABI, so we have to use it too
 if [ "$ARROW_USE_OLD_CXXABI" ]; then
-  $PKG_CFLAGS="$PKG_CFLAGS -D_GLIBCXX_USE_CXX11_ABI=0"
+  PKG_CFLAGS="$PKG_CFLAGS -D_GLIBCXX_USE_CXX11_ABI=0"
+fi
+
+if [ "$ARROW_R_CXXFLAGS" ]; then
```

Comment:
Add a note to the README about this feature?

Reply:
Will do